Lib/test/test_traceback.py

```diff
@@ -1187,6 +1188,215 @@ def test_traceback_header(self):
         self.assertEqual(list(exc.format()), ["Exception: haven\n"])
 
 
+class TestTracebackGroupException(unittest.TestCase):
+
+    def test_simple_exception(self):
```

Comment:
I broke up this method and had a more careful look at the tests. Some of the method I copied from TracebackException tests actually don't need to be applied to TracebackExceptionGroup because what they tests is not the TracebackExceptionGroup  API.

For simple exceptions, it's enough to test that:
1. TracebackExceptionGroup(e).summary[0] is equal to TracebackException(e)
2. The APIs exposed in TracebackExceptionGroup (from_exception, format, format_exception_only, str) are working as expected.

I wanted to factor out the TracebackException.format/format_exception_only tests so they can be shared, but then foudn that they do not actually exist (these methods are tested via the module level functions, which after this PR will be calling the TracebackExceptionGroup mothods, which call the TracebackException methods). 

So I think there are some improvements that can be made to the current test_traceback module, and maybe it makes sense to do them in a separate PR against master (so it can be merged irrespective of this one).

1. Some of your comment on this PR are for copies that I have now deleted, but they are relevant to the old version of the same test as well.
2. Add tests for the TracebackException.* methods directly
3. test_comparison for TracebackException is missing the case of non-equal TracebackException objects (it only has comparisons between equal TracebackExceptions, or a TracebackException and some other type).

Reply:
Yeah if I change TracebackException.__eq__ to do:
```
        if isinstance(other, TracebackException):
            return True
        return NotImplemented
```

then all the tests pass.